client/hooks/useMyViewport.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react';
+
+const useMyViewport = () => {
+  const [width, setWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const handleWindowResize = () => setWidth(window.innerWidth);
+    window.addEventListener('resize', handleWindowResize);
+    return () => window.removeEventListener('resize', handleWindowResize);
+  }, []);
+
+  return { width };
+};
```

Comment:
If we're using this hook for calculating the media query breakpoint, we should consider using the  [window.matchMedia](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) API instead of the `window.innerWidth` property. This will allow us to handle different screen sizes and resolutions more performantly. The `resize` event fires on every pixel change and can be resource-intensive, check out [this article](https://tigeroakes.com/posts/resize-event-alternatives/) for more.  We could do something like this: 

```

export default function useMediaQuery(query) {
  const subscribe = React.useCallback((callback) => {
    const matchMedia = window.matchMedia(query); 

    matchMedia.addEventListener('change', callback); 
  
    return () => {
      matchMedia.removeEventListener("change", callback)
    }
  },[query])

const getSnapshot = () => {
  return window.matchMedia(query).matches
}

const getServerSnapshot = () => {
  throw new Error('useMediaQuery only runs on client')
}

  return React.useSyncExternalStore(subscribe, getSnapshot, getServerSnapshot)
}
```
You can then use the hook like this which will return a boolean value based on the media query:
```
const isMobile = useMediaQuery("(max-width: 600px)")

  const isSmallDevice = useMediaQuery("only screen and (max-width : 768px)");
  const isMediumDevice = useMediaQuery(
    "only screen and (min-width : 769px) and (max-width : 992px)"
  );
  const isLargeDevice = useMediaQuery(
    "only screen and (min-width : 993px) and (max-width : 1200px)"
  );
  const isExtraLargeDevice = useMediaQuery(
    "only screen and (min-width : 1201px)"
  );

Reply:
@colinvandergraaf Great suggestion. I've been experimenting with it this morning and will implement.

Note: This application is not currently designed to be mobile friendly.